harper-core/src/currency.rs

```diff
@@ -0,0 +1,57 @@
+use is_macro::Is;
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Is, Clone, Copy, Serialize, Deserialize, PartialEq, Eq, PartialOrd, Hash)]
+pub enum Currency {
+    // $
+    Dollar,
+    // €
+    Euro,
+    // ₽
+    Ruble,
+    // ₺
+    Lira,
+    // £
+    Pound,
+    // ¥
+    Yen,
+    // ฿
+    Baht,
+    // ₩
+    Won,
+    // ₭,
+    Kip,
+}
+
+impl Currency {
+    pub fn from_char(c: char) -> Option<Self> {
+        let cur = match c {
+            '$' => Self::Dollar,
+            '€' => Self::Euro,
+            '₽' => Self::Ruble,
+            '₺' => Self::Lira,
+            '£' => Self::Pound,
+            '¥' => Self::Yen,
+            '฿' => Self::Baht,
+            '₩' => Self::Won,
+            '₭' => Self::Kip,
+            _ => return None,
+        };
```

Comment:
Maybe you would consider adding them all

https://en.wikipedia.org/wiki/Currency_Symbols_(Unicode_block)